conformance/tests/tuples_type_compat.py

```diff
@@ -71,15 +71,15 @@ def func4(
 def func5(val: tuple[int] | tuple[str, str] | tuple[int, *tuple[str, ...], int]):
     if len(val) == 1:
         # Type can be narrowed to tuple[int].
-        assert_type(val, tuple[int])  # tuple[int]
+        assert_type(val, tuple[int])  # E?: tuple[int]
```

Comment:
This makes the test not very useful (without manual review) since type checkers will now pass no matter what type they infer for `val`. One solution could be to use two types and asserts that exactly one of them errors, by using an error group and asserting for both the narrowed and non-narrowed type. However, we don't currently check in that case that there is exactly one error. I can send a PR doing that.

Reply:
Sounds reasonable to me.